### PR TITLE
manifest: ensure /var/roothome is available when needed

### PR DESCRIPTION
**Pull request by mvo5**
Currently creating a customization for the root user does not work because `/var/roothome` may not be avaialble in the image. This commit ensures it is created if needed just like we do for `/var/home`.

Closes https://github.com/osbuild/bootc-image-builder/issues/143

[edit: to (end-to-end) test I pointed https://github.com/osbuild/bootc-image-builder/pull/363 to this branch]

### CODE REVIEW

File: pkg/manifest/raw_bootc.go [hunk 1]
```diff
@@ -148,16 +181,12 @@ func (p *RawBootcImage) serialize() osbuild.Pipeline {
 		groupsStage.Devices = devices
 		pipeline.AddStage(groupsStage)
 	}
+
 	if len(p.Users) > 0 {
-		// ensure /var/home is available
+		// ensure home root dir (currently /var/home, /var/roothome) is
+		// available
 		mkdirStage := osbuild.NewMkdirStage(&osbuild.MkdirStageOptions{
```
Comment (supakeen) [inline pkg/manifest/raw_bootc.go:188]: Can we skip the entire `osbuild.NewMkdirStage` if `buildHomedirPaths(...) == nil`?

Comment (mvo5) [reply pkg/manifest/raw_bootc.go:188]: Thanks, that is an interesting question. In practice `if len(p.Users) > 0` then there will always be a buildHomedirPath result that is also > 0 - but that is true today who knows about tomorrow so maybe worth adding (otoh, tradeoffs, it adds one more level of if nested that is currently not needed and even if we never update it the worst case would be that we have an mkdir stage that just does nothing) - sorry, not very helpful I suppose :)

Comment (supakeen) [reply pkg/manifest/raw_bootc.go:188]: As long as the first premise holds (we always have to create homedirs for users) it's fine to keep as is :)

**Review by supakeen — APPROVED** [review]

**Comment by achilleas-k** [comment]
Rebasing for unit test fixes.

**Review by achilleas-k — APPROVED** [review]

**Pull request merged**